app/src/ui/diff/side-by-side-diff.tsx

```diff
@@ -408,6 +411,89 @@ export class SideBySideDiff extends React.Component<
         }
       }
     }
+
+    if (this.state.lastExpandedHunk !== prevState.lastExpandedHunk) {
+      this.focusAfterLastExpandedHunkChange()
+    }
+  }
+
+  /**
+   * This handles app focus after a user has clicked on an diff expansion
+   * button. With the exception of the top expand up button, the expansion
+   * buttons disappear after clicking and by default the focus moves to the app
+   * body. This is not ideal for accessibilty as a keyboard user must then tab
+   * all the way back to the diff to continut to interact with it.
+   *
+   * If an expansion button of the type clicked is available, we focus it.
+   * Otherwise, we try to find the next closest expansion button and focus that.
+   * If no expansion buttons available, we focus the diff container. This makes
+   * it so if a user expands down and can expand down further, they will
+   * automatically be focused on the next expand down.
+   *
+   * Other context:
+   * - When a user clicks on a diff expansion button, the
+   * lastExpandedHunk state is updated. In the componentDidUpdate, we detect
+   * that change in order to call this after the new expansion buttons have
+   * rendered. The rendered expansion buttons are stored in a map.
+   * - A hunk index may have multiple expansion buttons (up and down) so it does
+   *   not uniquely identify a button.
+   */
+  private focusAfterLastExpandedHunkChange() {
+    if (this.state.lastExpandedHunk === null) {
+      return
+    }
+
+    const diffNode = findDOMNode(this.virtualListRef.current)
+    const diff = diffNode instanceof HTMLElement ? diffNode : null
```

Comment:
I'd love it if we could defer this lookup to when it's actually needed. Could we extract this into a `focusListElement` method and call that in the conditional below and at the end of the method?